### PR TITLE
clean up text implementation

### PR DESCRIPTION
**Pull request by SimonDanisch**
And fix CI

### CODE REVIEW

File: GLMakie/src/drawing_primitives.jl [hunk 1]
```diff
@@ -274,34 +280,24 @@ function draw_atomic(screen::GLScreen, scene::Scene,
...
-            collect_glyph_data = (projview, transfunc, pos, rotation, model, space, offset) -> begin
-                gcollection = glyphcollection[]
-                res = Vec2f0(widths(pixelarea(scene)[]))
-                preprojected_glyph_arrays(pos, gcollection, space, projview, res, offset, transfunc)
-            end
-            glyph_data = lift(collect_glyph_data, scene.camera.projectionview, Makie.transform_func_obs(scene), args...)
+            # and here we wrap it into another observable
+            # so it doesn't triggers dimension mismatches
+            # the actual, new value gets then taken in the below lift with to_value
+            gcollection = Observable(glyphcollection)
```
Comment (jkrumbiegel) [inline GLMakie/src/drawing_primitives.jl:289]: why Observable{Observable{Vector{GlyphCollection}}}?

File: GLMakie/src/drawing_primitives.jl [hunk 2]
```diff
@@ -274,34 +280,24 @@ function draw_atomic(screen::GLScreen, scene::Scene,
...
         else
-            # and here we don't because it triggers dimension mismatches
-            collect_glyph_data = (projview, transfunc, pos, rotation, model, space, offset) -> begin
-                gcollection = glyphcollection[]
-                res = Vec2f0(widths(pixelarea(scene)[]))
-                preprojected_glyph_arrays(pos, gcollection, space, projview, res, offset, transfunc)
-            end
-            glyph_data = lift(collect_glyph_data, scene.camera.projectionview, Makie.transform_func_obs(scene), args...)
+            # and here we wrap it into another observable
+            # so it doesn't triggers dimension mismatches
```
Comment (jkrumbiegel) [inline GLMakie/src/drawing_primitives.jl:287]: trigger

File: WGLMakie/src/particles.jl [hunk 3]
```diff
@@ -173,34 +175,23 @@ function create_shader(scene::Scene, plot::Makie.Text{<:Tuple{<:Union{<:Makie.Gl
...
     else
-        # and here we don't because it triggers dimension mismatches
-        collect_glyph_data = (projview, transfunc, pos, rotation, model, space, offset) -> begin
-            gcollection = glyphcollection[]
-            res = Vec2f0(widths(pixelarea(scene)[]))
-            Makie.preprojected_glyph_arrays(pos, gcollection, space, projview, res, offset, transfunc)
-        end
-        glyph_data = lift(collect_glyph_data, scene.camera.projectionview, Makie.transform_func_obs(scene), args...)
+        # and here we wrap it into another observable
+        # so it doesn't triggers dimension mismatches
```
Comment (jkrumbiegel) [inline WGLMakie/src/particles.jl:182]: same

**Review by jkrumbiegel — CHANGES_REQUESTED** [review]

File: GLMakie/src/drawing_primitives.jl (hunk 1 above)
Comment (SimonDanisch) [reply GLMakie/src/drawing_primitives.jl:289]: So one can pass the observable through `lift` ... This avoids using two different functions for those code paths

**Comment by ffreyer** [comment]
Does this fix text transformations / #1145?

**Comment by SimonDanisch** [comment]
Yeah, I think it should! Sorry, completely forgot about communicating that!

**Comment by t-bltg** [comment]
Great ! Thanks both of you :wink:

This is minor to me, but changing the text size in the mwe `ax3d[:ticks, :textsize] = 20 ...` pushes the text labels further than [before](https://github.com/JuliaPlots/Makie.jl/pull/1145#issue-692089012):
dist


@SimonDanisch maybe we should take 5 min here to write a text unittest  in https://github.com/JuliaPlots/Makie.jl/tree/master/ReferenceTests/src/tests (translation, rotation, scale) so that it does not happen again and got covered by CI ? Even though I don't see what could go in it at the moment ...

**Pull request merged**